### ISSUE TITLE
keras.layers.SimpleRNN/LSTM output NaN when setting "activation" to "exponential"

### ISSUE THREAD

**Issue by mazeltovlee** [keras-team/keras#15912]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.7.0
- Python version: 3.7.12
- Bazel version (if compiling from source): N/A
- GPU model and memory: N/A
- Exact command to reproduce: Please see below

**Describe the problem**.
`keras.layers.SimpleRNN` and `keras.layers.LSTM` output `NaN` when setting the `activation` parameter to `exponential` if the shape of input is larger than 3.

**Describe the current behavior**.
Please see above
**Describe the expected behavior**.
`keras.layers.SimpleRNN` and `keras.layer.LSTM` should not output `NaN` under this setting.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): no
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**.
```
from tensorflow import keras
x = keras.layers.Input((6,3))
buggy_layer = keras.layers.SimpleRNN(50, activation="exponential")
y = buggy_layer(x)
model = keras.Model(x, y)
import numpy as np
input = np.random.rand(10,6,3)
res = model.predict(input)
model_path = "simple_rnn_exponential.h5"
model.save(model_path)
print(res)
```
You may also access the code here:
https://colab.research.google.com/drive/1mVBIPF79kiIYggZUUkvs1IEB1GdxeoJC?usp=sharing

To help you identify this issue, I have tried several ways to construct the buggy layer.

Trial 1:
Use `linear` as the activation while adding an additional exponential activation after the SimpleRNN. In this way, the `res` will not output `NaN`
```
from tensorflow import keras
x = keras.layers.Input((6,3))
buggy_layer = keras.layers.SimpleRNN(50, activation="linear")
y = buggy_layer(x)
z = keras.activations.exponential(y)
model = keras.Model(x, z)
import numpy as np
input = np.random.rand(10,6,3)
res = model.predict(input)
print(res)
```

Trial 2:
I tried converting the generated model which will output `NaN` to PyTorch using ONNX, I find that: after converting to PyTorch, the output will contain `Inf` and `0` but no `NaN`. Please see this link for more details about this trial:
https://colab.research.google.com/drive/1mVBIPF79kiIYggZUUkvs1IEB1GdxeoJC?usp=sharing

**Comment by Cheril311** [comment]
Hey, I think Pytorch does not have any 'exponential' activation function, so when your code is being converted using onnx do you know which pytorch activation function is being used

**Comment by mazeltovlee** [comment]
> Hey, I think Pytorch does not have any 'exponential' activation function, so when your code is being converted using onnx do you know which pytorch activation function is being used

Yes, following the implementation of onnx2pytorch, the exponential activation will be converted to `torch.exp`.
https://github.com/ToriML/onnx2pytorch/blob/89a3fe7b208952760138c550c499fe9c57c35d45/onnx2pytorch/convert/operations.py#L115

Apart from this, may I know that SimpleRNN/LSTM output NaN when setting activation as `Exponential` is the expected behavior?

**Comment by Cheril311** [comment]
I think this is because of the **exploding gradients** problem as the gradients are increasing exponentially in your case. As technically you get NaNs when you perform computations that dont make sense, for example _inf-inf_. However I am not sure if getting those NaNs in your case is expected in place of Infs. I think Stack Overflow will be a better place to ask this.
